atlasdb-config/src/main/java/com/palantir/atlasdb/config/AtlasDbConfig.java

```diff
@@ -219,16 +230,14 @@ protected final void check() {
 
         Preconditions.checkState(lock().isPresent() == timestamp().isPresent(),
                 "Lock and timestamp server blocks must either both be present or both be absent.");
-        if (getSweepBatchSize() != null || getSweepCellBatchSize() != null) {
-            Preconditions.checkState(
-                    getSweepReadLimit() == null
-                            && getSweepCandidateBatchHint() == null
-                            && getSweepDeleteBatchHint() == null,
-                    "Your configuration mixes both the old and the new parameters"
-                            + " for setting sweep batch sizes. Please use 'sweepMaxCellTsPairsToExamine',"
-                            + " 'sweepCandidateBatchSize' and 'sweepDeleteBatchSize' instead of the deprecated"
-                            + " 'sweepBatchSize' and 'sweepCellBatchSize'.");
-        }
+
+        Preconditions.checkState(getSweepBatchSize() == null
```

Comment:
A bit awkward to have these deprecated, then fail to start if they are used. That said, looks like too much lift to have a cleaner solution.